tests/trace/test_dataset.py

```diff
@@ -54,3 +55,13 @@ def greet(name: str, age: int) -> str:
     assert rows[1]["inputs"]["name"] == "Bob"
     assert rows[1]["inputs"]["age"] == 25
     assert rows[1]["output"] == "Hello Bob, you are 25!"
+
+
+def test_dataset_caching(client):
+    ds = weave.Dataset(rows=[{"a": i} for i in range(200)])
+    ref = weave.publish(ds)
+
```

Comment:
do we want to write a test that detects whether or not there is that annoying ```logger.error(
                    f"Expected length of response rows ({len(response.rows)}) to match prefetched rows ({len(self._prefetched_rows)}). Ignoring prefetched rows."
                )``` message?